src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,129 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int ONE_ELEMENT = 1;
+    private static final double SIZE_GROWTH_INDEX = 1.5;
+
+    private Object[] elementData;
+    private int capacity = DEFAULT_CAPACITY;
+    private int size;
+
+    public ArrayList(int initialCapacity) {
+        if (initialCapacity >= 0 && initialCapacity < capacity) {
+            this.elementData = new Object[capacity];
+        } else if (initialCapacity > capacity) {
+            this.elementData = new Object[capacity];
+            this.elementData = grow();
+        } else {
+            throw new IllegalArgumentException("Illegal Capacity: "
+                    + initialCapacity);
+        }
+    }
+
+    public ArrayList() {
+        this.elementData = new Object[capacity];
+    }
+
     @Override
     public void add(T value) {
-
+        add(value,size);
```

Comment:
Add additional logic here instead of call another method:
- grow if need.
- add element by index.
- increase size

Reply:
done